src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -139,7 +139,7 @@ public virtual void ProcessBlock(Block block)
 
                 if (newTip.Height > this.walletTip.Height)
                 {
-                    ChainedBlock findTip = newTip.FindAncestorOrSelf(this.walletTip.HashBlock);
+                    ChainedBlock findTip = newTip.GetAncestor(this.walletTip.Height);
```

Comment:
this change is wrong, consider newTip being on chain that does not contain walletTip

Reply:
Yes if they aren't on the same chain this won't work. I'll revert it back to use FindAncestorOrSelf which does a linear search for hash.